narwhals/dataframe.py

```diff
@@ -330,10 +330,22 @@ def unpivot(
         variable_name: str | None,
         value_name: str | None,
     ) -> Self:
+        on_ = [on] if isinstance(on, str) else on
+        index_ = (
+            [index]
+            if isinstance(index, str)
+            else index
+            if isinstance(index, list)
+            else []
+        )
```

Comment:
this might be causing issues (see plotly failing test)?

Reply:
Alright, I just pushed the variable and value name parsing on the BaseFrame level. `index` and `on` require parsing only for arrow and duckdb anyway. I think it's fine to keep it at the compliant level